sensors/SensorCommon.py

```diff
@@ -0,0 +1,40 @@
+from abc import ABC, abstractmethod
+import numpy as np
+from gym import spaces
+
+
+def Dist2Circ(x1, y1, x2,  y2, r):
+    dx = x2 - x1
+    dy = y2 - y1
+    Dist2Center = (dx ** 2 + dy ** 2) ** (1 / 2)
+    if Dist2Center < r:
+        return np.zeros(2)
+    dx = dx * (Dist2Center - r) / Dist2Center
+    dy = dy * (Dist2Center - r) / Dist2Center
+    return dx, dy
+
+
+class Sensor(ABC):
+
+    def __init__(self, nbObservations, SensorRange=10):
+        self._nbObservations = nbObservations
+        self._limSensor = SensorRange
+        self._Observation = []
+
+    @abstractmethod
+    def name(self):
+        pass
+
+    def getOSpaceSize(self):
+        return self._nbObservations, 2
+
+    def getObservationSpace(self):
+        return spaces.Box(low=-self._limSensor, high=self._limSensor, shape=self.getOSpaceSize(), dtype=np.float64)
+
+    @abstractmethod
+    def _reset(self):
```

Comment:
More out of interest: Are you following some specific guide (PEP rules for example) or why are some methods started with `_` (underscore)?

Reply:
My idea behind that was to make methods, which are meant for internal use only (kind of like private methods), starting with `_` (underscore). Is this the same way you are using the `_` in the codebase? Would be nice to stay consistent with your style. 

Also, do you think the _reset() methods for the sensors should be only for internal use or also external?

Reply:
The issue is that I have no consistent style at the moment. I am style looking for one. :D
We should start looking into PEP8 guidelines as it might help.

Concerning this specific point: You can keep it as `_reset()` and then we could clean up once we settled on one style guide.

Reply:
sounds good to me 😄